Mancala.hs

```diff
@@ -31,36 +31,92 @@ mancala_start = State (side_start, side_start) (actions_start, actions_start)
 --mancala move (State ((my_pockets, my_store), (other_pockets, other_store)) (my_actions, other_actions))
 -- - move pieces
 
+--precondition: index >= 0
 get_val :: Board -> Int -> Int
```

Comment:
seems like the naming convention is camelCase in haskell. I'll change them later.